### PR TITLE
Off-chain message signing - compatibility fixes

### PR DESCRIPTION
**Pull request by 0xMMBD**
#### Problem
Current version of the off-chain message signing was not fully compliant with the specification.

#### Summary of Changes
* Fixed support for off-chain message signing
* Application domain param validation moved to input_validators 
* Added tests for string_utils and odd-chain messages

### CODE REVIEW

File: cli/src/wallet.rs [hunk 1]
```diff
@@ -280,105 +281,112 @@ impl WalletSubCommands for App<'_, '_> {
...
-                )
-                .arg(pubkey!(
-                    Arg::with_name("signer")
+                        "Source account of funds (if different from client local account). "),
+                    )
+                    .arg(
+                        Arg::with_name("no_wait")
+                            .long("no-wait")
+                            .takes_value(false)
+                            .help("Return signature immediately after submitting the transaction, instead of waiting for confirmations"),
```
Comment (jnwng) [inline cli/src/wallet.rs:290]: are these just whitespace changes?

**Review by jnwng — COMMENTED** [review]
thanks for including Trent's changes!

File: cli/src/wallet.rs (hunk 1 above)
Comment (0xMMBD) [reply cli/src/wallet.rs:290]: yes, git presents it a little bit weird

**Comment by 0xMMBD** [comment]
@jnwng @t-nelson Hello, Do you have any additional feedback for this code? It was auto closed by a bot

**Pull request closed without merging**